manifest.go

```diff
@@ -613,31 +633,37 @@ func WriteManifestList(out io.Writer, files []ManifestFile) error {
 		}
 	}
 
-	var key string
+	var sch avro.Schema
 	switch version {
 	case 1:
-		key = internal.ManifestListV1Key
+		sch = internal.MustNewManifestListV1Schema()
 	case 2:
-		key = internal.ManifestListV2Key
+		sch = internal.MustNewManifestListV2Schema()
 	default:
 		return fmt.Errorf("%w: non-recognized version %d", ErrInvalidArgument, version)
 	}
```

Comment:
for the ManifestList schema and others that don't need the dynamic part, it's better to have a cache of the schemas than to build it from scratch each time.